src/box/replication.cc

```diff
@@ -632,8 +711,56 @@ replicaset_update(struct applier **appliers, int count, bool keep_connect)
 				  "duplicate connection to the same replica");
 		}
 		replica_hash_insert(&uniq, replica);
+		if (replicaset_state != REPLICASET_JOIN)
+			continue;
+		int ballot_size = applier->ballot.registered_replica_uuids_size;
+		struct tt_uuid *uuids =
+			applier->ballot.registered_replica_uuids;
+		for (int i = 0; i < ballot_size; i++) {
+			struct tt_uuid *uuid = &uuids[i];
+			for (int j = 0; j < uuid_count; j++) {
+				if (tt_uuid_compare(uuid, &registered_uuids[j]) == 0)
+					goto next;
+			}
+			registered_uuids[uuid_count++] = *uuid;
+			if (uuid_count >= VCLOCK_MAX) {
+				say_warn("Too many registered replicas discovered");
+				tnt_raise(ClientError,
+					  ER_BOOTSTRAP_CONNECTION_NOT_TO_ALL);
+			}
+next:
+		; /* nop */
+		}
 	}
 
+	int required_count = uuid_count;
+	if (self_position == -1)
```

Comment:
`self_position` is not updated in the loop above. This condition is always true. Same about the next usage.

Reply:
Oh. Yep, must be an artifact of some rebase. Sorry.
Anyway, given your comment about "should we demand that replica connects to itself", let's remove this logic altogether and only force the replica to connect to other registered peers:
```diff
diff --git a/src/box/replication.cc b/src/box/replication.cc
index cb06c6366..efb31c77c 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -678,7 +678,6 @@ replicaset_update(struct applier **appliers, int count, bool keep_connect)
 
        struct tt_uuid registered_uuids[VCLOCK_MAX];
        int uuid_count = 0;
-       int self_position = -1;
        /* Check for duplicate UUID */
        for (int i = 0; i < count; i++) {
                applier = appliers[i];
@@ -731,11 +730,8 @@ next:
                }
        }
 
-       int required_count = uuid_count;
-       if (self_position == -1)
-               required_count += 1;
        if (replicaset_state == REPLICASET_JOIN &&
-           required_count > count) {
+           uuid_count > count) {
                say_warn("Not all replica set members listed in "
                         "box.cfg.replication");
                struct replica key;
@@ -746,13 +742,6 @@ next:
 
                        say_warn("No connection to %s", tt_uuid_str(&key.uuid));
                }
-               if (self_position == -1) {
-                       key.uuid = INSTANCE_UUID;
-                       if (replica_hash_search(&uniq, &key) == NULL) {
-                               say_warn("No connection to %s",
-                                        tt_uuid_str(&key.uuid));
-                       }
-               }
                if (bootstrap_strategy == BOOTSTRAP_STRATEGY_AUTO &&
                    !replication_anon) {
                        tnt_raise(ClientError,
```